models/resnet/resnet.hpp

```diff
@@ -0,0 +1,485 @@
+   * @param downSampleInputHeight Input height for down-sample block.
+   * @param kernelWidth Width of the filter/kernel.
+   * @param kernelHeight Height of the filter/kernel.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   */
+  template <typename AddMergeType = ann::AddMerge<>>
+  void DownSample(AddMergeType* resBlock,
+                  const size_t inSize,
+                  const size_t outSize,
+                  const size_t downSampleInputWidth,
+                  const size_t downSampleInputHeight,
+                  const size_t kernelWidth = 1,
+                  const size_t kernelHeight = 1,
+                  const size_t strideWidth = 2,
+                  const size_t strideHeight = 2,
+                  const size_t padW = 0,
+                  const size_t padH = 0)
+  {
+    ann::Sequential<>* downSampleBlock = new ann::Sequential<>();
+    ConvolutionBlock(downSampleBlock, inSize, outSize, strideWidth,
+        strideHeight, kernelWidth, kernelHeight, padW, padH, true,
+        downSampleInputWidth, downSampleInputHeight);
+
+    resBlock->Add(downSampleBlock);
+  }
+
+  /**
+   * Adds basicBlock block for ResNet 18 and 34.
+   *
+   * It's represented as:
+   * 
+   * /code
+   * resBlock - AddMerge layer
+   * {
+   *   sequentialBlock - sequentialLayer
+   *   {
+   *     ConvolutionBlock(inSize, outSize, strideWidth, strideHeight)
+   *     ReLU
+   *     ConvolutionBlock(inSize, outSize)
+   *   }
+   *
+   *   sequentialLayer
+   *   {
+   *     if downsample == true
+   *       ConvolutionBlock(inSize, outSize, downSampleInputWidth,
+   *           downSampleInputHeight)
+   *     else
+   *       IdentityLayer
+   *   }
+   * 
+   *   ReLU
+   * }
+   * \endcode
+   * 
+   * @param inSize Number of input maps.
+   * @param outSize Number of output maps.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   * @param downSample If there will be a downSample block or not, default
+   *     false.
+   */
+  void BasicBlock(const size_t inSize,
+                  const size_t outSize,
+                  const size_t strideWidth = 1,
+                  const size_t strideHeight = 1,
+                  const bool downSample = false)
+  {
+    downSampleInputWidth = inputWidth;
+    downSampleInputHeight = inputHeight;
+
+    ann::Sequential<>* basicBlock = new ann::Sequential<>();
+    ann::AddMerge<>* resBlock = new ann::AddMerge<>(true, true);
+    ann::Sequential<>* sequentialBlock = new ann::Sequential<>();
+    ConvolutionBlock(sequentialBlock, inSize, outSize, strideWidth,
+        strideHeight);
+    sequentialBlock->Add(new ann::ReLULayer<>);
+    mlpack::Log::Info << "Relu" << std::endl;
```

Comment:
Lets create a function for this, since ReLU is being added a couple of times through out the code, and similarly logs are also being added.